app/views/layouts/angular-bootstrap/_auth_credentials_angular_bootstrap.html.haml

```diff
@@ -15,28 +15,33 @@
 - cancel_password_change ||= _("Cancel password change")
 - verify_title_off       ||= _("Server information, Username and matching password fields are needed to perform verification of credentials")
 - ng_show_userid         ||= true
+- guid_regex             ||= false
 
 %div{"ng-controller" => "CredentialsController"}
   %div{"ng-show" => "#{ng_show} && #{ng_show_userid}"}
     .form-group{"ng-class" => "{'has-error': angularForm.#{prefix}_userid.$invalid}"}
       %label.col-md-2.control-label{"for" => "#{prefix}_userid"}
         = userid_label
       .col-md-4
-        %input.form-control{"type"                    => "text",
-                            "id"                      => "#{prefix}_userid",
-                            "ng-required"             => "#{ng_reqd_userid}",
-                            "ng-disabled"             => userid_disabled,
-                            "name"                    => "#{prefix}_userid",
-                            "ng-model"                => "#{ng_model}.#{prefix}_userid",
-                            "checkchange"             => "",
-                            "ng-trim"                 => false,
-                            "detect_spaces"           => "",
-                            "prefix"                  => "#{prefix}",
-                            "reset-validation-status" => "#{prefix}_auth_status"}
+        - hash_for_userid = {"type"                    => "text",
+                             "id"                      => "#{prefix}_userid",
+                             "ng-required"             => "#{ng_reqd_userid}",
+                             "ng-disabled"             => userid_disabled,
+                             "name"                    => "#{prefix}_userid",
+                             "ng-model"                => "#{ng_model}.#{prefix}_userid",
+                             "checkchange"             => "",
+                             "ng-trim"                 => false,
+                             "detect_spaces"           => "",
+                             "prefix"                  => "#{prefix}",
+                             "reset-validation-status" => "#{prefix}_auth_status"}
+        - hash_for_userid['is-uuid'] = 4 if guid_regex
```

Comment:
The reason I picked `4` was because one of the other JS libraries enforced the version to be  `4` (which I thought was the standard, but maybe I was mistaken)
So I'll change the `'is-uuid'` value to `'all'`, meaning we'll be supporting all UUID versions...unless I hear otherwise. cc @djberg96 @bronaghs